### PR TITLE
Add SampleMask Array Limit VUID

### PR DESCRIPTION
**Pull request by sfricke-samsung**
From inactive #1950 

I just removed the clip/control part as still need to address @jeffbolznv comments, but the `SampleMask` built in seemed fine and rather get something in then wait until I get around to fix the clip/control part

VUID-VkPipelineShaderStageCreateInfo-maxSampleMaskWords-00711

### CODE REVIEW

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -1864,6 +1864,47 @@ bool CoreChecks::ValidatePushConstantUsage(const PIPELINE_STATE &pipeline, SHADE
...
+                                       VkShaderStageFlagBits stage) const {
+    bool skip = false;
+
+    // Currently all builtin tested are only found in fragment shaders
+    if (stage != VK_SHADER_STAGE_FRAGMENT_BIT) {
+        return skip;
+    }
+
+    for (auto id : accessible_ids) {
+        auto insn = src->get_def(id);
```
Comment (ncesario-lunarg) [inline layers/shader_validation.cpp:1877]: `const` (not 100% sure if the returned iterator will allow this to be const)?

File: layers/shader_validation.cpp [hunk 2]
```diff
@@ -1864,6 +1864,47 @@ bool CoreChecks::ValidatePushConstantUsage(const PIPELINE_STATE &pipeline, SHADE
...
+        decoration_set decorations = src->get_decorations(insn.word(2));
```
Comment (ncesario-lunarg) [inline layers/shader_validation.cpp:1878]: Looks like this value also does not change?

**Review by ncesario-lunarg — COMMENTED** [review]
With the possible exception of one minor change and the addition of some `const` qualifiers, this LGTM and is passing internal CI.

File: layers/shader_validation.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply layers/shader_validation.cpp:1877]: ya, this one won't allow it, but will fix the others

**Review by locke-lunarg — APPROVED** [review]
LGTM ! Wait for all CI green !

**Comment by ncesario-lunarg** [comment]
Running updates through internal CI.

**Review by ncesario-lunarg — APPROVED** [review]
Passing internal CI and LGTM. Waiting for travis to get going again.

**Comment by ncesario-lunarg** [comment]
Created #2283 to try and get this through travis.

**Pull request merged**